src/shims/foreign_items/posix/macos.rs

```diff
@@ -0,0 +1,116 @@
+        let this = self.eval_context_mut();
+
+        match link_name {
+            "__error" => {
+                let errno_place = this.machine.last_error.unwrap();
+                this.write_scalar(errno_place.to_ref().to_scalar()?, dest)?;
+            }
+
+            // File related shims
+             "open" => {
+                let result = this.open(args[0], args[1])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
+            }
+
+            "close$NOCANCEL" => {
+                let result = this.close(args[0])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
+            }
+
+            "stat$INODE64" => {
+                let result = this.stat(args[0], args[1])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
+            }
+
+            "lstat$INODE64" => {
+                let result = this.lstat(args[0], args[1])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
+            }
+
+            "fstat$INODE64" => {
+                let result = this.fstat(args[0], args[1])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
+            }
+
+            "lseek" => {
+                let result = this.lseek64(args[0], args[1], args[2])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
+            }
+
+            // Time related shims
+            "gettimeofday" => {
+                let result = this.gettimeofday(args[0], args[1])?;
+                this.write_scalar(Scalar::from_int(result, dest.layout.size), dest)?;
+            }
+
+            // macOS API stubs.
+            "pthread_attr_get_np" => {
+                this.write_null(dest)?;
+            }
+
+            "pthread_get_stackaddr_np" => {
+                let stack_addr = Scalar::from_uint(STACK_ADDR, dest.layout.size);
+                this.write_scalar(stack_addr, dest)?;
+            }
+
+            "pthread_get_stacksize_np" => {
+                let stack_size = Scalar::from_uint(STACK_SIZE, dest.layout.size);
+                this.write_scalar(stack_size, dest)?;
+            }
+
+            "_tlv_atexit" => {
+                // FIXME: register the destructor.
+            }
+
+            "_NSGetArgc" => {
+                this.write_scalar(this.machine.argc.expect("machine must be initialized"), dest)?;
+            }
+
+            "_NSGetArgv" => {
+                this.write_scalar(this.machine.argv.expect("machine must be initialized"), dest)?;
+            }
+
+            "SecRandomCopyBytes" => {
+                let len = this.read_scalar(args[1])?.to_machine_usize(this)?;
+                let ptr = this.read_scalar(args[2])?.not_undef()?;
+                this.gen_random(ptr, len as usize)?;
+                this.write_null(dest)?;
+            }
+
+            "syscall" => {
```

Comment:
Oh you duplicated this between Linux and macOS? Hm... that's a bit odd I feel.

Reply:
the main reason is that `statx` is a linux-only shim.

Reply:
Hm. The code duplication is not great though.

Can you make this its own dispatch function? With a common base case in posix.rs, then dispatching to linux and macos for more target-specific syscalls?

Reply:
But also, are you sure this macos code here is reachable? This interface strikes me as very linux-specific.

Reply:
I don't know why I thought that macos used this api too, but I couldn't find any hard evidence on it.